compiler/ccgtypes.nim

```diff
@@ -282,7 +282,7 @@ proc ccgIntroducedPtr(conf: ConfigRef; s: PSym, retType: PType): bool =
     result = false
   # first parameter and return type is 'lent T'? --> use pass by pointer
   if s.position == 0 and retType != nil and retType.kind == tyLent:
-    result = pt.kind != tyVar
+    result = pt.kind notin {tyVar, tyArray}
```

Comment:
But C arrays are pointers already, why would the C code generator introduce another indirection in this case?

Reply:
The fix here doesn't add double pointer, it removes it.

Reply:
But then what about ``tyOpenArray, tyVarargs`` (and maybe ``tySequence``)?

Reply:
Yes, you are right tyOpenArray, tyVarargs also need this fix.
Added tests to cover tyOpenArray, tyVarargs to PR

Reply:
Also, big sets are mapped to C arrays... :-)

Reply:
Yes, they are and they need the fix. Added them as well :)